src/service/viewer-impl.js

```diff
@@ -802,6 +802,25 @@ export class Viewer {
   }
 
   /**
+   * Get the fragment from the url or the viewer.
+   * Strip leading '#' in the fragment
+   * @return {!Promise<string|undefined>}
```

Comment:
That's because we have to trust the viewer for it to return a CID. With this, trust isn't an issue, and it'll allow us to [cleanup](https://github.com/ampproject/amphtml/pull/4019#discussion_r72492830).